### DISCUSSION TITLE
Cancelar ou excluir eventos?

### DISCUSSION THREAD

**Discussion by jenrodrigues** [andrebaltieri/AgendaEventos#6] [Q&A] [answered]
No nosso product backlog nao esta claro o que EXCLUIR um evento significa. 
- O evento vai ser excluido da base de dados?
- O evento vai permanecer na base de dados com um status Cancelado?

**Comment by renebentes** [comment]
Penso que faz mais sentido cancelar um evento.

**Comment by LuanFreitasRibeiro** [comment]
Acredito que podemos dar as duas opções. Faz sentido? O user pode tanto excluir o evento, quanto cancelar.

**Reply by renebentes** [reply]
Exclusão apenas se não existir participantes no evento?

**Reply by jenrodrigues** [reply]
Pensei em "exclusao" caso o Organizador ou o Admin tenham inserido um evento que por algum motivo nao deveria estar na base, nem mesmo constando como Cancelado. E o unico caso no qual eu consigo pensar...

**Comment by BrewertonSantos** [comment]
Exclusão em caso de, por exemplo: Evento criado de forma errada. E entregar a possíbilidade de exclusão além de apenas editar é útil do meu ponto de vista. Mas ainda assim Exclusão seria um termo abstrato já que a nível de db essa informação pode ser utilizada de diversas formas. Então não seria excluida realmente. Apenas desabilitada.

**Reply by LeonardoPedrosa** [reply]
Acredito que esse seja o caminho correto a seguir.

**Reply by nilson-teodoro** [reply]
Gostei da sugestão Brewerton, e que tal ter exclusão lógica? Nesse caso, não seria um delete, podemos ter uma coluna deletedAt informando a data em que foi deletado, qual usuário fez a ação, podendo ser organizador ou administrador, se fizer sentido...

**Comment by jenrodrigues** [comment]
Talvez deixar para o Organizador somente a opcao de CANCELAR um evento, e para o Administrador a opcao de EXCLUIR o evento? O que acham?

**Reply by LeonardoPedrosa** [reply]
Acredito que é uma solução bem viável e aceitável.

**Comment by jenrodrigues** [comment, marked as answer]
Foi decidido que o usuario do tipo Organizador so podera CANCELAR um evento. Se, em algum momento, decidirmos implementar a EXCLUSAO de um evento, achamos conveniente restringir esse tipo de atividade para usuarios do tipo Administrador.
